src/main/java/net/spy/memcached/MemcachedClient.java

```diff
@@ -280,19 +285,19 @@ private MemcachedClient(ConnectionFactory cf, List<InetSocketAddress> addrs, boo
     transcoder = cf.getDefaultTranscoder();
     opFact = cf.getOperationFactory();
     assert opFact != null : "Connection factory failed to make op factory";
-
-    if(clientMode == ClientMode.Dynamic){
-      initializeClientUsingConfigEndPoint(cf, addrs.get(0));
-    } else {
-      setupConnection(cf, addrs);
-    }
-
+    
     operationTimeout = cf.getOperationTimeout();
     authDescriptor = cf.getAuthDescriptor();
     executorService = cf.getListenerExecutorService();
     if (authDescriptor != null) {
       addObserver(this);
     }
+    
+    if(clientMode == ClientMode.Dynamic){
+      initializeClientUsingConfigEndPoint(cf, addrs.get(0));
+    } else {
+      setupConnection(cf, addrs);
+    }
```

Comment:
Can you move this above to before line 292 above? i.e. before the `addObserver()` logic and try to see if that addresses the 5 seconds of client initialization logic in dynamic mode for TLS? 

That way you can keep the fix for NPE inside `addObserver()` for SASL authentication when

Reply:
Sure. Will give it a try